sklearn/preprocessing/data.py

```diff
@@ -1719,10 +1726,15 @@ def _transform_selected(X, transform, selected="all", copy=True):
         X_sel = transform(X[:, ind[sel]])
         X_not_sel = X[:, ind[not_sel]]
 
-        if sparse.issparse(X_sel) or sparse.issparse(X_not_sel):
-            return sparse.hstack((X_sel, X_not_sel))
+        if order:
+            # As of now, X is expected to be dense array
```

Comment:
I just tried this. Apparently, `X[:, ind[sel]] = X_sel` is not implemented for csc matrices.

```